public/app/core/services/backend_srv.ts

```diff
@@ -175,15 +175,33 @@ export class BackendSrv implements BackendService {
     return this.dependencies.fromFetch(url, init).pipe(
       mergeMap(async response => {
         const { status, statusText, ok, headers, url, type, redirected } = response;
-        const textData = await response.text(); // this could be just a string, prometheus requests for instance
         let data: T;
 
-        try {
-          data = JSON.parse(textData); // majority of the requests this will be something that can be parsed
-        } catch {
-          data = textData as any;
+        if (options.responseType) {
+          switch (options.responseType) {
+            case 'arraybuffer':
+              data = (await response.arrayBuffer()) as any;
+              break;
+            case 'blob':
+              data = (await response.blob()) as any;
+              break;
+            case 'json':
+              data = await response.json();
+              break;
+            case 'text':
+              data = (await response.text()) as any;
+              break;
+          }
+        }
+        // @ts-ignore
+        if (!data) {
```

Comment:
@hugohaggmark any suggestions for a more clean approach?

Reply:
@ryantxu I would try to use a handler/strategy pattern [like we do for headers](https://github.com/grafana/grafana/blob/b2dcf06b6017aab7719ccf3a8e4de389c9b6c1b7/public/app/core/utils/fetch.ts#L59).

We could also turn this into separate RxJs operators. That would be super neat now that I think about it more, but handler pattern would do perfectly too and we already use it.

Reply:
@hugohaggmark can you take another look?  Is this what you were thinking?

Reply:
Hmm, not really what I meant, but this is a great start!